mila_datamodules/vision/datasets/bch_coco.py

```diff
@@ -0,0 +1,98 @@
+import json, time
+from typing import List
+
+from PIL import Image
+from torchvision.datasets import CocoDetection
+
+import bcachefs as bch
+
+
+class BchCocoDetection(CocoDetection):
+    """`MS Coco Detection <https://cocodataset.org/#detection-2016>`_ Dataset.
+
+    It requires the `COCO API to be installed <https://github.com/pdollar/coco/tree/master/PythonAPI>`_.
+
+    Args:
+        cursor (bch.Cursor): Bcachefs filesystem cursor.
+        annFile (string): Path to json annotation file.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.PILToTensor``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        transforms (callable, optional): A function/transform that takes input sample and its target as entry
+            and returns a transformed version.
+    """
+
+    def __init__(
+        self,
+        cursor: bch.Cursor,
+        annFile: str,
+        *args, **kwargs
+    ) -> None:
+        CocoDetection.__init__(self, root=cursor.filename, annFile=None, *args,
```

Comment:
Personal preference on my end but I'll let the final choice to you. `super()` could mean that this `class` is safe to use with and has bean thought to work with mixins or multiple classes heritage. In this case, very few torchvision's dataset are safe to use in that context even if not explicitly warned so I suppose it can be assumed that this heritage would not be safe anyway

Reply:
(but the suggestion would break the code as it would duplicate the `**kwargs)` part)